src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -203,6 +207,63 @@ select new
             }
         }
 
+        /// <summary>
+        ///     Update Longest Build Path table once a day for each channel at midnight
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [CronSchedule("0 0 0 1/1 * ? *", TimeZones.PST)]
+        public async Task UpdateLongestBuildPathAsync(CancellationToken cancellationToken)
+        {
+            IQueryable<Channel> query = Context.Channels;
+
+            List<Channel> channels = query.AsEnumerable().Select(c => new Channel() { Id = c.Id, Name = c.Name }).ToList();
+
+            // Get the flow graph
+            var barOnlyRemote = await RemoteFactory.GetBarOnlyRemoteAsync(Logger);
+
+            List<Microsoft.DotNet.Maestro.Client.Models.DefaultChannel> defaultChannels = (await barOnlyRemote.GetDefaultChannelsAsync()).ToList();
+            List<Microsoft.DotNet.Maestro.Client.Models.Subscription> subscriptions = (await barOnlyRemote.GetSubscriptionsAsync()).ToList();
+
+            IEnumerable<string> frequencies = 
+                new string[] { "everyWeek", "twiceDaily", "everyDay", "everyBuild", "none", };
+
+            foreach (var channel in channels)
+            {
+                // Build, then prune out what we don't want to see if the user specified
+                // channels.
+                DependencyFlowGraph flowGraph = await DependencyFlowGraph.BuildAsync(defaultChannels, subscriptions, barOnlyRemote, 30);
+
+                flowGraph.PruneGraph(
+                    node => DependencyFlowGraph.IsInterestingNode(channel.Name, node), 
+                    edge => DependencyFlowGraph.IsInterestingEdge(edge, false, frequencies));
+
+                if (flowGraph.Nodes.Count > 0)
+                {
+
+                    flowGraph.MarkBackEdges();
+                    flowGraph.CalculateLongestBuildPaths();
+                    flowGraph.MarkLongestBuildPath();
+
+                    // Get the nodes on the longest path and order them by path time so that the
+                    // contributing repos are in the right order
+                    IEnumerable<DependencyFlowNode> longestBuildPathNodes = flowGraph.Nodes.Where(n => n.OnLongestBuildPath).OrderByDescending(n => n.BestCasePathTime);
+
+                    LongestBuildPath lbp = new LongestBuildPath()
+                    {
+                        Channel = channel,
+                        BestCaseTimeInMinutes = longestBuildPathNodes.Max(n => n.BestCasePathTime),
+                        WorstCaseTimeInMinutes = longestBuildPathNodes.Max(n => n.WorstCasePathTime),
+                        ContributingRepositories = String.Join(';', longestBuildPathNodes.Select(n => $"{n.Repository}@{n.Branch}").ToArray()),
```

Comment:
>ContributingRepositories = String.Join(';', longestBuildPathNodes.Select(n => $"{n.Repository}@{n.Branch}").ToArray()), [](start = 24, length = 119)

This is just to display to users right (so it's ok if it's maybe got a bug in it)?  If we are using it for logic anywhere, we shouldn't do string manipulations, and should just have a many-to-many mapping table.

Reply:
Yes, this is just for users. It'll be a tool tip in the power bi report